### PR TITLE
feat: remove foxy rebase history

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- removes rebase functionality from investor-foxy and web consumption
- removes the notion of rebase from web `useBalanceChartData` and `txHistorySlice`
- removes now unused Infura env var

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/5736

## Risk

- EVM accounts (and consequently, net worth) balances may be borked, test accordingly

## Testing

- Ensure FOXy price is still happy 
- Ensure FOXy DeFi opportunity, user data (balances) and functionality (deposit/withdraw/claim) is still happy
- Ensure EVM accounts' balances looks sane against prod
- Ensure net worth looks sane against prod

### Engineering

- ☝🏽 
- grep for rebase and ensure all runtime occurences are gone

### Operations

- ☝🏽 

### FOXy balance

- Prod

Screenshot 2023-12-05 at 11 35 19

- This diff

Screenshot 2023-12-05 at 11 35 27

### Net worth

- Prod

Screenshot 2023-12-05 at 11 34 54

### FOXy DeFi

- This diff

image

- Prod

image

- This diff

 
Screenshot 2023-12-05 at 11 35 00

### Perf

- Prod

Screenshot 2023-12-05 at 11 46 30
Screenshot 2023-12-05 at 11 46 23

Notes:

1.  in prod, Infura and *our* JSON-RPC node requests are decoupled. That's because develop isn't whitelisted for Infura, so we use our node as a bandaid currently, which makes FOxy unreliable in localhost currently
2. JSON-RPC requests are batched, and prod being faster, the batching may be more efficient, so the number of requests in prod vs. here for the same account isn't really indicative of anything

- This diff

Screenshot 2023-12-05 at 11 44 46

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5777** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5777?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5777?utm_source=stack-comment).

File: src/state/slices/txHistorySlice/txHistorySlice.ts [hunk 1]
```diff
@@ -218,44 +208,10 @@ export const txHistory = createSlice({
...
-        const foxyApi = getFoxyApi()
-
-        ;(async () => {
-          const rebaseHistoryArgs = { userAddress, tokenContractAddress: foxyTokenContractAddress }
-          const data = await foxyApi.getRebaseHistory(rebaseHistoryArgs)
-          const assetReference = foxyTokenContractAddress
-          const assetNamespace = ASSET_NAMESPACE.erc20
-          const assetId = toAssetId({ chainId, assetNamespace, assetReference })
-          const upsertPayload = { accountId, assetId, data }
-          if (data.length) dispatch(txHistory.actions.upsertRebaseHistory(upsertPayload))
```
Comment (gomesalexandre) [inline src/state/slices/txHistorySlice/txHistorySlice.ts:254]: `upsertRebaseHistory` too

Comment (gomesalexandre) [reply src/state/slices/txHistorySlice/txHistorySlice.ts:254]: fc6e77061f

File: src/lib/investor/investor-foxy/api/api.ts [hunk 2]
```diff
@@ -1088,129 +1086,4 @@ export class FoxyApi {
       throw new Error(`Failed to get information from Tokemak ipfs ${e}`)
     }
   }
-
-  async getRebaseHistory(input: BalanceInput) {
```
Comment (woodenfurniture) [inline src/lib/investor/investor-foxy/api/api.ts:1092]: 🩸

**Review by woodenfurniture — APPROVED** [review]
first pass stamp

**Pull request merged**